SourceryTests/ParserSpec.swift

```diff
@@ -119,13 +119,128 @@ class ParserSpec: QuickSpec {
+                            .to(equal([Type(name: "Foo", methods: [])]))
+                    }
+
+                    it("extracts method properly") {
+                        expect(parse("class Foo { func bar(some: Int) ->  Bar {}; func foo() ->    Foo {}; func fooBar() ->Foo }")).to(equal([
+                            Type(name: "Foo", methods: [
+                                Method(fullName: "bar(some:)", parameters: [
+                                    Method.Parameter(name: "some", typeName: "Int")
+                                    ], returnTypeName: "Bar"),
+                                Method(fullName: "foo()", returnTypeName: "Foo"),
+                                Method(fullName: "fooBar()", returnTypeName: "Foo")
+                                ])
+                        ]))
+                    }
+                    
+                    it("extracts class method properly") {
+                        expect(parse("class Foo { class func foo() }")).to(equal([
+                            Type(name: "Foo", methods: [
+                                Method(fullName: "foo()", parameters: [], isClass: true)
+                                ])
+                            ]))
+                    }
+
+                    it("extracts static method properly") {
+                        expect(parse("class Foo { static func foo() }")).to(equal([
+                            Type(name: "Foo", methods: [
+                                Method(fullName: "foo()", isStatic: true)
+                                ])
+                            ]))
+                    }
+
+                    context("given method with parameters") {
+                        it("extracts method with single parameter properly") {
+                            expect(parse("class Foo { func foo(bar: Int) }")).to(equal([
+                                Type(name: "Foo", methods: [
+                                Method(fullName: "foo(bar:)", parameters: [
+                                    Method.Parameter(name: "bar", typeName: "Int")])
+                                    ])
+                            ]))
+                        }
+                        
+                        it("extracts method with two parameters properly") {
+                            expect(parse("class Foo { func foo(bar: Int, foo: String) }")).to(equal([
+                                Type(name: "Foo", methods: [
+                                    Method(fullName: "foo(bar:foo:)", parameters: [
+                                        Method.Parameter(name: "bar", typeName: "Int"),
+                                        Method.Parameter(name: "foo", typeName: "String")
+                                        ], returnTypeName: "Void")
+                                    ])
+                            ]))
+                        }
+                        
+                        it("extracts method with parameter with two names") {
```

Comment:
I guess we will need a separate improvement for dealing with closures, like having `isClosure` property, list of closure parameters types, escaping and so on.